Automattic/jetpack
# AI Assistant: improve the process of getting post content

This PR improves the process of getting post content.
Currently, the approach relies on the `content` block attribute, which doesn't scale properly.
In this PR, the process has two steps:

* Serialize the blocks
* Transform the serialized content to markdown.

The second step removes the serialized part on the fly since, in the end, they are comments from the HTML perspective.

Fixes https://github.com/Automattic/jetpack/issues/30619

## Proposed changes:

* AI Assistant: improve the process of getting post content

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Add content to the post by using different block types: headers, paragraphs, lists, event images (set the image caption, for instance)
* Create an AI Assistant block instance
* Request an action to will consider the post content, for instance, translate:
* Confirm the prompt contains all content independently of the block type
* Confirm that the prompt can handle images

https://github.com/Automattic/jetpack/assets/77539/6f6333c9-8e58-48d5-bf57-0cf4286367e0

<img width="788" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/8b079128-a0a7-49ae-846c-cf166fbe972c">
